Python/marshal.c

```diff
@@ -921,6 +931,59 @@ r_float_str(RFILE *p)
     return PyOS_string_to_double(buf, NULL, NULL);
 }
 
+static PyObject *
+r_bytecode(RFILE *p)
+{
+    long size = r_long(p);
+    if (PyErr_Occurred()) {
+        return NULL;
+    }
+    Py_ssize_t nbytes = size * sizeof(_Py_CODEUNIT);
+    if (nbytes < 0 || SIZE32_MAX < nbytes) {
+        const char *e = "bad marshal data (bytecode size out of range)";
+        PyErr_SetString(PyExc_ValueError, e);
+        return NULL;
+    }
+    PyObject *bytecode = PyBytes_FromStringAndSize(NULL, nbytes);
+    if (bytecode == NULL) {
+        return NULL;
+    }
+    _Py_CODEUNIT *buffer = (_Py_CODEUNIT *)PyBytes_AS_STRING(bytecode);
+    long i = 0;
+    while (i < size) {
+        int opcode = r_byte(p);
+        if (opcode == EOF) {
+            const char *e = "EOF read where opcode expected";
+            PyErr_SetString(PyExc_EOFError, e);
+            return NULL;
+        }
+        int oparg;
+        if (HAS_ARG(opcode)) {
+            oparg = r_byte(p);
+            if (oparg == EOF) {
+                const char *e = "EOF read where oparg expected";
+                PyErr_SetString(PyExc_EOFError, e);
+                return NULL;
+            }
+        }
+        else {
+            oparg = 0;
+        }
+        assert(0x00 <= opcode && opcode < 0x100);
+        assert(0x00 <= oparg && oparg < 0x100);
+        buffer[i++] = _Py_MAKECODEUNIT(opcode, oparg);
+        for (int j = 0; j < _PyOpcode_Caches[opcode]; j++) {
+            buffer[i++] = _Py_MAKECODEUNIT(CACHE, oparg);
```

Comment:
Maybe assert that `i < size` here, since there is potential for a buffer overrun here.